### DISCUSSION TITLE
fly sync from v5.8.1 to v6.3.0 issue

### DISCUSSION THREAD

**Discussion by maxknee** [concourse/concourse#5848] [Help & Support]
## Summary
When upgrading to 6.3.0 from 5.8.1, running `fly sync`

```bash
WARNING:

fly version (5.8.1) is out of sync with the target (6.3.0). to sync up, run the following:
```
but getting an error after I log in

```bash
error: user [] is not in team [name]
```

## Steps to reproduce

fly sync 

asks me to login, and successfully logs in, but fails

## Expected results

fly sync upgrades successfully 

## Actual results

`error: user [] is not in team [team-1]`

## Triaging info

* Concourse version: 6.3.0 (fly 5.8.1)
* Browser (if applicable):  Chrome
* Did this used to work? (i think so?)

**Comment by jamieklassen** [comment]
@maxknee there is a release note describing this backwards incompatibility -- https://github.com/concourse/concourse/releases/tag/v6.1.0#4950. We don't have an exhaustive upgrade matrix and I don't know if we want to commit to it. That said, what kind of documentation do you think would have led you to re-download fly?

**Reply by jamieklassen** [reply]
you can click the CLI icons in the UI or you can curl `/api/v1/cli` manually to do this, since you don't have the `fly sync` without `-t` which was introduced in v6.0.

**Reply by maxknee** [reply]
Gotcha, we have a pretty large instance of concourse and we'll have to work on something to help our users upgrade from 5.8 to 6.3
unless would going from 5.8 -> 6.0 -> 6.1 -> 6.3 facilitate an easier upgrade path? 
A note indicating that a direct download would be necessary due to the fact that it is incompatible, would probably help

**Reply by jamieklassen** [reply]
you can do 5.8 -> 6.0 and `fly sync` with no errors, then 6.0 -> 6.3 will break but you can use `fly sync` without `-t` so you don't need to direct download. where would you expect to see an instruction to do a direct download? that's the tricky part because i don't really want to duplicate this advice for every post-6.1 release, y'know?

**Reply by jamieklassen** [reply]
basically 6.0 introduces a solution (syncing without having a target) that should solve this problem for all future upgrades

**Reply by maxknee** [reply]
understood, we held off to 6.0 upgrade initially waiting for issues to be addressed, and when we were ready we're just going from 5.8.1 to 6.3.0 and we'll have to figure out a way to make this easier for our users.
